### DISCUSSION TITLE
previously-working hald-clut now very dark

### DISCUSSION THREAD

**Discussion by teridon** [ImageMagick/ImageMagick#6173] [Help] [answered]
### ImageMagick version

7.1.1-3

### Operating system, version and so on

win10 21H2

### Description

It's been many years since I've done this, but it used to work ( see http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=27377 ).  In fact, if I go back to my old computer (with version 6.9.0), it still works.  But with 7.1.1-3, with the same input files and arguments, the output image is almost too dark to see. 

I created the color map using the process outlined in the thread above (i.e. applying a ggr to a hald:8 png) 

Example command-line:
`magick convert 2023_03_17__00_00_45_350__SDO_AIA_AIA_171.jp2 sdo-aia-171-colormap.png -hald-clut  output.jpg`

example colormap and output attached.  I can't upload the input image because it is JP2 and github will not let me upload it
input image: [2023_03_17__00_00_09_353__SDO_AIA_AIA_171.jp2](http://jsoc.stanford.edu/data/aia/images/2023/03/17/171/2023_03_17__00_00_09_353__SDO_AIA_AIA_171.jp2 )

6.9 output: 
![output-6 9 0](https://user-images.githubusercontent.com/1752560/226108246-c868375c-9319-41b0-b244-424ac0b7cf91.jpg)
7.1 output: 
![output-7 1 1](https://user-images.githubusercontent.com/1752560/226108247-32a342ab-d031-4afe-9b46-371baef7f868.jpg)
colormap ( sdo-aia-171-colormap.png ): 
![sdo-aia-171-colormap](https://user-images.githubusercontent.com/1752560/226108267-3d15836a-430e-4980-8ee9-60dd54be6657.png)

**Comment by fmw42** [comment]
If you want to upload, zip it first, then upload the zip file.

On IM 7, use magick, not magick convert and not convert.

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.

**Comment by teridon** [comment]
just for documentation purposes, the patch is here:  https://github.com/ImageMagick/ImageMagick/commit/852a723f1ea8a4dbf204f013173eb4e17ce4f0c1 

I compiled from git on Ubuntu and can confirm that the patch resolves the dark image issue.
